### DISCUSSION TITLE
Problem with tuning the PID

### DISCUSSION THREAD

**Discussion by martinmikula** [tttapa/Control-Surface-Motor-Fader#11] [Q&A]
Hello, first of all thank you very much for great library. I have an problem with tuning the PID. It seems that my main problem is that I'm trying to increase the Kp to have an stable oscillation, and in your example you have an stable oscillation in Kp = 19, I have an stable oscillation only when I changed it to 60... Next i measured the Tu, and count all values, but the problem is that I have still a really big overshoot... My suspection is that the main problem is related with the too high Kp... Do you have any idea what can be wrong? Bacically when I enable PWM with value 40 (max256) the fader is not moving at all, then I increasing the value (or help with my finger) then the fader starts to move really quickly...  Do you have any idea?

**Comment by tttapa** [comment]
It sounds like your fader has high static friction. This is a common problem with PID controllers, and there is a ton of research about it.

One workaround is to add feedforward control with a model that includes the friction. In the simplest case, you add some extra force when the velocity is close to zero:
```cpp
if (std::abs(velocity) < threshold)
  output += std::copysign(F, error);
```
You'll have to tune `threshold` and `F`.
